In [1]:
# Your task in this week’s assignment is to answer three questions:
# 1. What is the northernmost airport in the United States?
# 2. What is the easternmost airport in the United States?
# 3. On February 12th, 2013, which New York area airport had the windiest weather?

# Being successful on this assignment requires using both:
# • Your developing skills in Python and pandas, and
# • Your research skills in validating your candidate answers in context.

# Tasks
# 1. Use the information in the .CSV files found at
# https://github.com/hadley/nycflights13/tree/master/data-raw for your source data.
# 2. Write your code using Python and pandas, and deliver it in a Jupyter notebook. You should post
# your notebook in a GitHub repository, and send me a link to the notebook.
# 3. For each of the three questions listed above, you’ll need to verify your answers (in two cases by
# searching for additional confirming information, and in the third case being alert for possible
# outliers). For the first two questions above, you may want to start by generating a list of the five
# northernmost or easternmost U.S. airports.
# 4. In addition to providing code, you should state your answers to each of the above questions (in
# Markdown) in the Jupyter notebook.
# 5. To receive full credit, you’ll need to include in your Jupyter notebook both your code and your
# documented assumptions and any changes that you made to the data based on your research
# (using Markdown) in your Jupyter notebook.

# Saar Turjeman
# IS362 - assignment# 5
# SPRING2024

In [2]:
%pylab inline
import pandas as pd
from pandas import Series, DataFrame

Populating the interactive namespace from numpy and matplotlib


In [3]:
# import the airlines DataFrame, downloaded from GitHub
airlines = pd.read_csv('airlines.csv')

# import the airports DataFrame, downloaded from GitHub
airports = pd.read_csv('airports.csv')

# import the planes DataFrame, downloaded from GitHub
planes = pd.read_csv('planes.csv')

# import the weather DataFrame, downloaded from GitHub
weather = pd.read_csv('weather.csv')

# case 1

# Latitude is the primary determinant of northness or southness. The higher the latitude, the more north the location is

In [4]:
# 1. What is the northernmost airport in the United States?


# displays the airports DataFrame, and sorting the values in lat in descending order
# and shows the top 5 values using the .head() method
airports.sort_values('lat', ascending=False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

# case 2

# Checking based on Longitude 

# Longitude (lon): Longitude primarily indicates the east-west position of a location and does not directly relate to its northness. However, if latitude were the same between two locations, the one with the less negative longitude (or more positive) would be further east and therefore considered more north in relation to the same latitude.

In [5]:
airports.sort_values('lon', ascending=False).head()

faa                           name        lat         lon   alt  tz dst  \
1290  SYA                   Eareckson As  52.712275  174.113620    98  -9   A   
942   MYF               Montgomery Field  32.475900  117.759000    17   8   A   
396   DVT  Deer Valley Municipal Airport  33.411700  112.457000  1478   8   A   
417   EEN        Dillant Hopkins Airport  72.270833   42.898333   149  -5   A   
444   EPM     Eastport Municipal Airport  44.910111  -67.012694    45  -5   A   

                  tzone  
1290  America/Anchorage  
942      Asia/Chongqing  
396      Asia/Chongqing  
417                 NaN  
444    America/New_York

# case 3 - outlier


In [34]:
# Select latitudes where the latitude value is greater than or equal to the mean plus one standard deviation
airports['lat'][airports['lat'] >= airports['lat'].mean() + airports['lat'].std()]


16      54.013333
34      60.866667
65      63.329444
86      67.106389
97      57.749967
          ...    
1416    60.335278
1418    56.116389
1419    60.939167
1434    59.301200
1440    64.301203
Name: lat, Length: 242, dtype: float64

In [ ]:
# the answer for question # 1 - BRW
# 1. What is the northernmost airport in the United States?
sorted_airports = airports.sort_values('lat', ascending=False)
northernmost_airport = sorted_airports.iloc[1:2]
northernmost_airport

In [ ]:
# 2. What is the easternmost airport in the United States?

# Easternmost Airport:
# The higher the longitude the more east the airport is located.

# Sorting the values in the lon table showing the first five airports with the highest longitude


# case 1
### For the lon 174.113620: Despite being a positive longitude, it represents a location more westward, likely in the western hemisphere.
### For -67.0129694: Being a negative longitude, it signifies a location west of the Prime Meridian, making it more eastward compared to the positive longitude value provided. Therefore, -67.0129694 (likely in Maine) is more easternmost compared to 174.113620.


In [ ]:
airports.sort_values('lon', ascending=False).head(5)

# case 2
### I wanted to look by tz or tzone, but could not find all the cities listed. 
### missing cities from eastern states!

In [ ]:
airports.sort_values('tz', ascending=False).head(5)

# case 3

In [35]:
# Select longitude where the latitude value is greater than or equal to the mean plus one standard deviation
airports['lon'][airports['lon'] >= airports['lon'].mean() + airports['lon'].std()]

15     -71.196771
29     -70.839444
50     -72.864694
91     -70.060181
150    -71.514778
          ...    
1447   -72.681500
1449   -72.593054
1451   -73.541493
1452   -71.075000
1454   -72.925992
Name: lon, Length: 74, dtype: float64

# answer to question# 1

In [ ]:
# the answer for question # 1 - BRW
# 1. What is the northernmost airport in the United States?
sorted_airports = airports.sort_values('lat', ascending=False)
northernmost_airport = sorted_airports.iloc[1:2]
northernmost_airport

# answer to question# 2

In [ ]:
# the answer for question # 2 - EPM
# 1. What is the Easternmost airport in the United States?
sorted_airports = airports.sort_values('lon', ascending=False)
easternmost_airport = sorted_airports.iloc[4:5]
easternmost_airport

In [ ]:
# 3. On February 12th, 2013, which New York area airport had the windiest weather?

In [ ]:
weather.sort_values('wind_speed',ascending=False).head(5)

In [ ]:
feb12_2013 = weather[(weather['month'] == 2) & (weather['day'] == 12) & (weather['year'] == 2013)]
feb12_2013.sort_values('wind_speed', ascending=False).head(5)

# answer to question# 3

In [ ]:
# highest wind speed 
feb12_2013.sort_values('wind_speed', ascending=False).head(1)